drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1200 @@
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param samples - Number of samples to be returned
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl313_iio_read_samples(void* dev, int* buff, uint32_t samples)
+{
+	int16_t data_x;
+	int16_t data_y;
+	int16_t data_z;
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	for(uint32_t i = 0; i < samples*iio_adxl313->no_of_active_channels;) {
+		adxl313_get_raw_xyz(adxl313, &data_x, &data_y, &data_z);
+
+		if (iio_adxl313->active_channels & NO_OS_BIT(0)) {
+			buff[i] = no_os_sign_extend32(data_x, 12);
+			i++;
+		}
+		if (iio_adxl313->active_channels & NO_OS_BIT(1)) {
+			buff[i] = no_os_sign_extend32(data_y, 12);
+			i++;
+		}
+		if (iio_adxl313->active_channels & NO_OS_BIT(2)) {
+			buff[i] = no_os_sign_extend32(data_z, 12);
+			i++;
+		}
+	}
+
+	return samples;
+}
+
+/***************************************************************************//**
+ * @brief Updates the number of active channels and the total number of
+ * 		  active channels
+ *
+ * @param dev  - The iio device structure.
+ * @param mask - Mask of the active channels
+ *
+ * @return ret - Result of the updating procedure.
+*******************************************************************************/
+static int adxl313_iio_update_channels(void* dev, uint32_t mask)
+{
+	struct adxl313_iio_dev *iio_adxl313;
+	uint8_t counter = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	iio_adxl313->active_channels = mask;
+
+	if (mask & NO_OS_BIT(0))
+		counter ++;
+	if (mask & NO_OS_BIT(1))
+		counter ++;
+	if (mask & NO_OS_BIT(2))
+		counter ++;
+	if (mask & NO_OS_BIT(3))
+		counter ++;
+
+	iio_adxl313->no_of_active_channels = counter;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Handles trigger: reads one data-set and writes it to the buffer.
+ *
+ * @param dev_data  - The iio device data structure.
+ *
+ * @return ret - Result of the handling procedure. In case of success, the size
+ * 				 of the written data is returned.
+*******************************************************************************/
+static int32_t adxl313_trigger_handler(struct iio_device_data *dev_data)
+{
+	int32_t data_buff[3];
+	int16_t x,y,z;
+	uint8_t i = 0;
+
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev_data)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev_data->dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	adxl313_get_raw_xyz(adxl313, &x, &y, &z);
```

Comment:
use return value

Reply:
Modified in all occurrences of the function.